### DISCUSSION TITLE
Support passing local files to `scene.stl()`, `scene.gltf()` and `scene.texture()`

### DISCUSSION THREAD

**Discussion by javierlopezrodriguez** [zauberzeug/nicegui#3693] [Ideas / Feature Requests]
### Description

**Description:**

I've encountered an issue where loading an STL file works as expected when using `app.add_static_files`, but fails when trying to load the file after selecting it with the local file picker from the nicegui example. I need to be able to load the STL file dynamically after the user selects it (because I don't necessarily know its location), but the second approach does not work.

**Code Examples:**

1. **Working Example:**

```python
from nicegui import app, ui
from pathlib import Path
import os

local_path = Path.cwd() / "example_data"

app.add_static_files('/data', local_path)

with ui.scene(grid=False, width=1024, height=800, background_color="black") as scene:
    body_mesh = scene.stl('/data/body_mesh.stl').material(side="double")
    scene.move_camera(147,159,279)

ui.run()
```

- In this example, I'm using `app.add_static_files` to include the folder into the app, and the STL file is loaded into the scene without issues.

2. **Non-Working Example:**

```python
from nicegui import app, ui
from pathlib import Path
import os
from local_file_picker import local_file_picker 
# https://github.com/zauberzeug/nicegui/blob/main/examples/local_file_picker/local_file_picker.py

async def pick_file() -> None:
    result = await local_file_picker(Path.cwd()) # list
    local_path = result[0]
    assert os.path.exists(local_path)
    with scene:
        body_mesh = scene.stl(local_path).material(side="double")

@ui.page("/")
def main():
    global scene
    with ui.scene(grid=False, width=1024, height=800, background_color="black") as scene:
        scene.move_camera(147,159,279)

    with ui.row():
        ui.button("Load STL", on_click=pick_file)

ui.run()
```

- In this example, I'm allowing the user to select the file using a file picker. The path is obtained correctly, and the file exists at the given location, but the STL file fails to load when attempting to use it with `scene.stl(local_path)`.

The key difference between the two examples is how the file path is handled. In the first case, I'm using `app.add_static_files`, but in the second case, I rely on the user's selection and dynamically provide the path to `scene.stl()`. I need a solution that allows for dynamic loading of the STL file after user selection. Any guidance or workaround would be appreciated.

**Environment:**

- NiceGUI Version: 1.4.29
- Python Version: 3.12.4
- OS: Windows 10

**Comment by falkoschindler** [comment]
Hi @javierlopezrodriguez,

`scene.stl()` expects a URL, not a filepath. This is why the file can't be found without serving it as a static file. This is a bit different than, e.g., a `ui.image`, which automatically serves images given as local filepath.

We can convert this discussion into a feature request to automatically serve STL files as well. Alternatively you can let the `pick_file()` function temporarily copy the file into a folder of static files which is served via `app.add_static_files`. Then you can call `scene.stl` with a URL generated from the folder's remote path and the file name.

**Comment by javierlopezrodriguez** [comment]
Thank you! I missed the difference between the URL and the filepath requirements , I do not yet understand how app.add_static_files works and why that path counts as an URL. I managed to do it by copying the file into the shared folder, so I don't need anything else, but if you want to turn this into a discussion go ahead. Thanks again!
